drivers/serial/Kconfig

```diff
@@ -31,6 +31,20 @@ config SERIAL_SUPPORT_INTERRUPT
 	  This is an option to be enabled by individual serial driver
 	  to signal that the driver and hardware supports interrupts.
 
+config UART_USE_RUNTIME_CONFIGURE
+	bool "Enable runtime configuration for UART controllers"
+	default y
+	help
+	  Enable runtime configuration of UART controllers.
+	  This allows applications to call uart_configure() to
+	  configure the UART controllers at runtime, and calling
+	  uart_config_get() to retrieve configuration. If this is
+	  disabled, UART controllers rely on UART driver's
+	  initialization function to properly configure
+	  the controller.
+
```

Comment:
Maybe we can add something like "disabling will save footprint for applications that don't need this service" ?

Reply:
Added something similar.